### DISCUSSION TITLE
Confidence intervals

### DISCUSSION THREAD

**Discussion by MoBosse** [open-sdg/open-sdg#1789] [Ideas]
Recently, I had a case where a data provider supplied a dataset that had a confidence interval that should be displayed alongside the data.
Since I had no other idea, I modified the annotation lines a bit to make them work as a confidence interval.

![grafik](https://user-images.githubusercontent.com/50657629/186591403-6fc45c13-5718-4392-8b2e-c8779f998a24.png)

Although this worked fairly well, it comes with some drawbacks in that it is quite fiddly to implement, does not show up in either the downloadable data or the table, and does not look exactly like confidence intervals normally do (with those horizontal lines at the ends).

There are some approaches for confidence intervals on chartjs online, but I was not able to implement them.
I vaguely remember there being some interest in this from Iceland some time ago...? So maybe it would be worth to find a good solution here?

**Comment by brockfanning** [comment]
@MoBosse It looks like there is an active plugin for Chartjs: https://github.com/sgratzl/chartjs-chart-error-bars. Maybe we could integrate this with Open SDG.

However I think some extra work would need to be done to get something into the downloadable data and table. What is your impression for how this would be best represented in the table?

**Comment by MoBosse** [comment]
The plugin sounds good.
We currently have the confidence interval data simply attached as a PDF table in the "additional data" tab. So you could also do the realization in the chart detached from the tables.

About the presentation in the tables: I see two possible solutions here:
- Adding two separate columns with the original heading, followed by something like "(lower confidence limit)" or "(upper confidence limit)"
- Adding the lower and upper confidence limits in brackets behind/below the value inside the table cells.

The second solution sounds interesting to me, as we have long wanted to be able to add string content to the data to mark data points as "estimated" or "revised", etc. These flags could then be output in the table and in the tooltips in the chart and map. Perhaps this could go hand in hand with this feature?

**Comment by christoph-LSN** [comment]
@MoBosse @brockfanning Sounds interesting. I urgentliy need to add string content in a couple of my indicators. i've tried comment_OBS here: https://christoph-lsn.github.io/MT_Site/6-3-1/
 but this is not what I need... for @MoBosse I need it for the Mikrozensus tables... to do "Klammerungen" () and "Sperrungen" /
